src/openllm/models/baichuan/modeling_baichuan.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 BentoML Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+import typing as t
+
+import openllm
+
+
+if t.TYPE_CHECKING:
+    import torch
+
+    import bentoml
+    import transformers
+else:
+    torch = openllm.utils.LazyLoader("torch", globals(), "torch")
+    transformers = openllm.utils.LazyLoader("transformers", globals(), "transformers")
+
+
+class Baichuan(openllm.LLM["transformers.PreTrainedModel", "transformers.PreTrainedTokenizerBase"]):
+    __openllm_internal__ = True
+
+    def llm_post_init(self):
+        self.device = torch.device("cuda")
+
+    def load_model(self, tag: bentoml.Tag, *args: t.Any, **attrs: t.Any) -> t.Any:
+        trust_remote_code = attrs.pop("trust_remote_code", True)
+        return transformers.AutoModelForCausalLM.from_pretrained(
+            self.model_id, trust_remote_code=trust_remote_code, **attrs
+        )
+
+    def load_tokenizer(self, tag: bentoml.Tag, **attrs: t.Any) -> t.Any:
+        trust_remote_code = attrs.pop("trust_remote_code", True)
+        return transformers.AutoTokenizer.from_pretrained(self.model_id, trust_remote_code=trust_remote_code, **attrs)
+
+    def sanitize_parameters(
+        self,
+        prompt: str,
```

Comment:
From the quick look of it, it seems like some of the default models does support different prompt template. Can we so something similar to mpt prompt parsing here?

Reply:
Since the prompt template is kind of simple right now, I implemented a way like `falcon` and `flan_t5` do. Does it make sense to u?

Reply:
Ye I think the prompt can just be very simple for now, that's fine.